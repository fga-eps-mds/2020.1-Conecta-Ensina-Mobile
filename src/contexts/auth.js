import AsyncStorage from '@react-native-community/async-storage';
import React, {createContext, useState, useEffect} from 'react';
import {Alert} from 'react-native';
import {HOST} from '@env';

export const AuthContext = createContext({});

export default function AuthProvider({children}) {
  const [user, setUser] = useState(null);
  const [typeUser, setTypeUser] = useState(null);
  const [teacher, setTeacher] = useState(null);
  const [student, setStudent] = useState(null);
  const [loading, setLoading] = useState(true);

  const Host = `${HOST}`;

  useEffect(() => {
    async function loadStorage() {
      const storageTypeUser = await AsyncStorage.getItem('Auth_typeUser');
      const storageUser = await AsyncStorage.getItem('Auth_user');

      if (storageTypeUser) {
        if (JSON.parse(storageTypeUser) === 1) {
          setTypeUser('Adm');
          setLoading(false);
        } else if (JSON.parse(storageTypeUser) === 2) {
          setTypeUser('Professor');
          setLoading(false);
        } else {
          setTypeUser('Aluno');
          setLoading(false);
        }
      }
      if (storageUser) {
        const dataUser = JSON.parse(storageUser);
        let usuario = {
          id: dataUser.id,
          firstName: dataUser.firstName,
          lastName: dataUser.lastName,
          email: dataUser.email,
          password: dataUser.password,
          role: dataUser.role,
          cellphone: dataUser.cellphone,
        };
        setUser(usuario);
        setLoading(false);
        if (dataUser.role == 2 || dataUser.role == 3) {
          const storageStudent = await AsyncStorage.getItem('Auth_student');
          if (storageStudent) {
            const dataStudent = JSON.parse(storageStudent);
            let estudante = {
              birthdate: dataStudent.birthdate,
              grade: dataStudent.grade,
              institution: dataStudent.school,
              cpf: dataStudent.cpf,
              cep: dataStudent.cep,
              number: dataStudent.number,
              details: dataStudent.details,
              description: dataStudent.description,
              special: dataStudent.special,
            };
            setStudent(estudante);
            setLoading(false);
          }
          if (dataUser.role == 2) {
            const storageTeacher = await AsyncStorage.getItem('Auth_teacher');
            if (storageTeacher) {
              const dataTeacher = JSON.parse(storageTeacher);
              let professor = {
                photo: dataTeacher.photo,
                video: dataTeacher.video,
                graduation_area: dataTeacher.graduation_area,
                degree: dataTeacher.degree,
                bank: dataTeacher.bank,
                agency: dataTeacher.agency,
                account: dataTeacher.account,
              };
              setTeacher(professor);
              setLoading(false);
            }
          }
        }
      }
      setLoading(false);
    }
    loadStorage();
  }, []);

  async function signIn(email, password) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: email,
        password: password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/login`, settings);
    try {
      const data = await fetchResponse1.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Login efetuado com sucesso!') {
          console.log('login efetuado');
          storageTypeUser(data.role);
          renderData(data.id, data.role, password);
          if (data.role === 1) {
            setTypeUser('Adm');
            console.log('Adm');
          } else if (data.role === 2) {
            setTypeUser('Professor');
            console.log('Professor');
          } else {
            setTypeUser('Aluno');
            console.log('aluno');
          }
          setUser(data);
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
  } //close signIn function

  async function registerAluno(values, {setStatus}) {
    var ok = false;
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
      }),
    };
    const fetchResponse = await fetch(`${Host}/api/student/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Estudante criado com sucesso!') {
          ok = true;

          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Aluno');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(3);
    }
  } //close registerAluno fuction

  async function registerProf(values, {setStatus}) {
    var ok = false;
    console.log(
      'firstName: ' +
        values.name +
        '\n' +
        'lastName: ' +
        values.surname +
        '\n' +
        'email: ' +
        values.email +
        '\n' +
        'password: ' +
        values.password +
        '\n' +
        'cellphone: ' +
        values.cellphone +
        '\n' +
        'birthdate: ' +
        values.birthdate +
        '\n' +
        'grade: ' +
        values.grade +
        '\n' +
        'institution: ' +
        values.school +
        '\n' +
        'cpf: ' +
        values.cpf +
        '\n' +
        'cep: ' +
        values.cep +
        '\n' +
        'number: ' +
        values.num +
        '\n' +
        'details: ' +
        values.details +
        '\n' +
        'description: ' +
        values.description +
        '\n' +
        'special: ' +
        values.special +
        '\n' +
        'photo: ' +
        values.photo +
        '\n' +
        'video: ' +
        values.video +
        '\n' +
        'graduation_area: ' +
        values.graduation_area +
        '\n' +
        'degree: ' +
        values.degree +
        '\n' +
        'bank: ' +
        values.bank +
        '\n' +
        'agency: ' +
        values.agency +
        '\n' +
        'account: ' +
        values.account +
        '\n',
    );
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
        photo: values.photo,
        video: values.video,
        graduation_area: values.graduation_area,
        degree: values.degree,
        bank: values.bank,
        agency: values.agency,
        account: values.account,
      }),
    };

    const fetchResponse = await fetch(`${Host}/api/teacher/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Professor criado com sucesso!') {
          ok = true;
          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Professor');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(2);
    }
  } //close registerProf function

  const RegistroAlert = ({role}) => {
    Alert.alert(
      'Registro',
      'Registro Concluido com sucesso',
      [
        {
          text: 'Finalizar',
          onPress: () => {
            if (role === 2) {
              setTypeUser('Professor');
            } else {
              setTypeUser('Aluno');
            }
          },
        },
      ],
      {cancelable: false},
    );
  };

  async function renderData(id, role, password) {
    var ok = false;
    try {
      let response = await fetch(`${Host}/api/user/` + id);
      const data = await response.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      if (role === 2 || role === 3) {
        ok = true;
      }
      storageUser(usuario);
      setUser(usuario);
    } catch (error) {
      console.error(error);
    }
    if (ok) {
      try {
        let response = await fetch(`${Host}/api/student/` + id);
        const data = await response.json();
        let estudante = {
          birthdate: data.data.student.birthdate,
          grade: data.data.student.grade,
          institution: data.data.student.institution,
          cpf: data.data.student.cpf,
          cep: data.data.student.cep,
          number: data.data.student.number,
          details: data.data.student.details,
          description: data.data.student.description,
          special: data.data.student.special,
        };
        setStudent(estudante);
        storageStudent(estudante);
      } catch (error) {
        console.error(error);
      }
      if (role === 2) {
        try {
          let response = await fetch(`${Host}/api/teacher/` + id);
          const data = await response.json();
          let professor = {
            photo: data.data.teacher.photo,
            video: data.data.teacher.video,
            graduation_area: data.data.teacher.graduation_area,
            degree: data.data.teacher.degree,
            bank: data.data.teacher.bank,
            agency: data.data.teacher.agency,
            account: data.data.teacher.account,
          };
          setTeacher(professor);
          storageTeacher(professor);
        } catch (error) {
          console.error(error);
        }
      }
    }
  }

  async function updateUser(values, id) {
    const settings1 = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        cellphone: values.cellphone,
        password: user.password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/${id}`, settings1);

    try {
      const data = await fetchResponse1.json();
      console.log('Success');
    } catch (error) {
      console.error('Error:', error);
    }
    if (user.role === 2 || user.role === 3) {
      const settings2 = {
        method: 'PUT',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          agentRole: user.role,
          birthdate: student.birthdate,
          grade: values.grade,
          institution: values.school,
          cpf: student.cpf,
          cep: values.cep,
          number: values.num,
          details: values.details,
          description: values.description,
          special: student.special,
        }),
      };
      const fetchResponse2 = await fetch(
        `${Host}/api/student/${id}`,
        settings2,
      );
      try {
        const data2 = await fetchResponse2.json();
        console.log('Success');
      } catch (error) {
        console.error('Error:', error);
      }
      if (user.role === 2) {
        const settings3 = {
          method: 'PUT',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            agentRole: user.role,
            video: values.video,
            graduation_area: teacher.graduation_area,
            degree: values.degree,
            bank: values.bank,
            agency: values.agency,
            account: values.account,
          }),
        };
        const fetchResponse3 = await fetch(
          `${Host}/api/teacher/${id}`,
          settings3,
        );
        try {
          const data3 = await fetchResponse3.json();
          console.log('Success');
        } catch (error) {
          console.error('Error:', error);
        }
      }
    }
    renderData(user.id, user.role, user.password);
  }

  async function storageTypeUser(data) {
    await AsyncStorage.setItem('Auth_typeUser', JSON.stringify(data));
  }

  async function storageUser(data) {
    await AsyncStorage.setItem('Auth_user', JSON.stringify(data));
  }

  async function storageStudent(data) {
    await AsyncStorage.setItem('Auth_student', JSON.stringify(data));
  }

  async function storageTeacher(data) {
    await AsyncStorage.setItem('Auth_teacher', JSON.stringify(data));
  }

  async function signOut() {
    await AsyncStorage.clear().then(() => {
      setUser(null);
      setTypeUser(null);
      setStudent(null);
      setTeacher(null);
    });
  }

  return (
    <AuthContext.Provider
      value={{
        signed: !!user,
        user,
        loading,
        teacher,
        student,
        typeUser,
        Host,
        signIn,
        registerAluno,
        registerProf,
        updateUser,
        signOut,
      }}>
      {children}
    </AuthContext.Provider>
  );
}
