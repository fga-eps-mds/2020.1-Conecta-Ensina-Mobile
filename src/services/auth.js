export async function signIn(email, password, Host) {
  const settings = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      email: email,
      password: password,
    }),
  };
  const fetchResponse1 = await fetch(`${Host}/api/user/login`, settings);
  try {
    const data = await fetchResponse1.json();
    console.log('Success:', data);
    if (data.message) {
      if (data.message === 'Login efetuado com sucesso!') {
        console.log('login efetuado');
        return data;
        /*renderData(data.id, data.role);
        if (data.role === 1) {
          setTypeUser('Adm');
          console.log('Adm');
        } else if (data.role === 2) {
          setTypeUser('Professor');
          console.log('Professor');
        } else {
          setTypeUser('Aluno');
          console.log('aluno');
        }
        setUser(data);*/
      }
    }
  } catch (error) {
    console.error('Error:', error);
  }
} //close signIn function
