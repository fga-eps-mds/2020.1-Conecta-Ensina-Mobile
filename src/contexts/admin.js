import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
