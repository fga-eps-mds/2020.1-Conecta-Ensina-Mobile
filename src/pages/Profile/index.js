import React, {useContext, useState} from 'react';
import Theme from '../../../Theme';

import Background1 from '../../components/Background1';
import CustomText from '../../components/CustomText';
import ProfileFields from '../../components/ProfileFields';
import gradeResolver from '../../services/gradeResolver';

import {AuthContext} from '../../contexts/auth';

import {
  Container,
  InfoContainer,
  ButtonContainer,
  ContainerBotao,
} from './styles';

export default function Profile({navigation}) {
  const {user, student, signOut} = useContext(AuthContext);
  let infoList = [
    {
      text: user && user.firstName ? user.firstName : 'Não informado',
      field: 'Nome',
    },
    {
      text: user && user.lastName ? user.lastName : 'Não informado',
      field: 'Sobrenome',
    },
    {
      text: user && user.email ? user.email : 'Não informado',
      field: 'Email',
    },
    {
      text: user && user.cellphone ? user.cellphone : 'Não informado',
      field: 'Celular',
    },
    {
      text: student && student.cep ? student.cep : 'Não informado',
      field: 'CEP',
    },
    {
      text: student && student.number ? student.number : 'Não informado',
      field: 'Número do endereço',
    },
    {
      text: student && student.details ? student.details : 'Não informado',
      field: 'Complemento do endereço',
    },
    {
      text: student && student.cpf ? student.cpf : 'Não informado',
      field: 'CPF',
    },
    {
      text:
        student && student.grade
          ? gradeResolver(student.grade)
          : 'Não informado',
      field: 'Grau de estudo',
    },
    {
      text:
        student && student.description ? student.description : 'Não informado',
      field: 'Descrição',
    },
    {
      text:
        student && student.institution ? student.institution : 'Não informado',
      field: 'Instituição de estudo',
    },
  ];
  return (
    <Theme>
      <Background1 navigation={navigation}>
        <Container>
          <InfoContainer>
            {infoList.map((item, key) => (
              <ProfileFields key={key} field={item.field}>
                {item.text}
              </ProfileFields>
            ))}
          </InfoContainer>
        </Container>
        <ContainerBotao>
          <ButtonContainer
            testID="Editar"
            onPress={() => navigation.navigate('Edit')}>
            <CustomText white bigSmall>
              Editar
            </CustomText>
          </ButtonContainer>
          <ButtonContainer testID="Sair" onPress={() => signOut()}>
            <CustomText white bigSmall>
              Sair
            </CustomText>
          </ButtonContainer>
        </ContainerBotao>
      </Background1>
    </Theme>
  );
}
