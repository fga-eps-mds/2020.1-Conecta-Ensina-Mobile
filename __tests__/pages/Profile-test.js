import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {AuthContext} from '../../src/contexts/auth';
import Profile from '../../src/pages/Profile';

describe('Testing Profile page', () => {
  const student = {
    cep: '',
    number: '',
    details: '',
    cpf: '',
    description: '',
    institution: '',
  };
  const user = {
    firstName: '',
    lastName: '',
    email: '',
    cellphone: '',
  };
  const navigation = {
    navigate: jest.fn(),
  };
  const signOut = jest.fn();

  test('Should matches snapshot', () => {
    const tree = render(
      <AuthContext.Provider value={{signOut, user, student}}>
        <Profile navigation={navigation} />
      </AuthContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
  test("Should press 'Editar' button", () => {
    const {getByTestId} = render(<Profile navigation={navigation} />);
    const button = getByTestId('Editar');

    fireEvent.press(button);
  });

  test("Should press 'Sair' button", () => {
    const {getByTestId} = render(
      <AuthContext.Provider value={{signOut}}>
        <Profile navigation={navigation} />
      </AuthContext.Provider>,
    );

    const button = getByTestId('Sair');

    fireEvent.press(button);
  });
});
