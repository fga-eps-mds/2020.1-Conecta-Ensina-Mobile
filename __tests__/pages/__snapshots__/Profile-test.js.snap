// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Profile page Should matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 22,
        "flexShrink": 1,
        "marginLeft": 30,
        "opacity": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 1000,
          "height": 58,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginLeft": 40,
          "opacity": 1,
          "width": 58,
        }
      }
      testID="userAvatar"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/user_blue.png",
          }
        }
        style={
          Array [
            Object {
              "height": 38,
              "width": 33,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 40,
            "marginRight": 35,
          },
        ]
      }
    >
      <Text
        medium={true}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 25,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#184E79",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 4,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 100,
            "justifyContent": "center",
            "left": 0,
            "right": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "center",
          }
        }
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "width": "100%",
            },
          ]
        }
      >
        <View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Nome
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sobrenome
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Email
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Celular
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            CEP
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Número do endereço
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Complemento do endereço
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            CPF
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Grau de estudo
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Descrição
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
          <Text
            black={true}
            medium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "textAlign": "center",
                },
              ]
            }
          >
            Instituição de estudo
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#184E79",
                  "borderRadius": 9,
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 280,
                },
              ]
            }
          >
            <Text
              medium={true}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "textAlign": "center",
                  },
                ]
              }
              white={true}
            >
              Não informado
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 0.15,
            "flexShrink": 1,
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E46270",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 140,
          }
        }
        testID="Editar"
      >
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Editar
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E46270",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 140,
          }
        }
        testID="Sair"
      >
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Sair
        </Text>
      </View>
    </View>
  </View>
</View>
`;
